staging/src/k8s.io/apiserver/pkg/storageversion/manager.go

```diff
@@ -146,6 +149,28 @@ func (s *defaultManager) UpdateStorageVersions(kubeAPIServerClientConfig *rest.C
 	s.setComplete()
 }
 
+// dedupResourceInfos dedups ResourceInfos with the same underlying storage.
+// ResourceInfos from the same Group with different Versions share the same underlying storage.
+// ResourceInfos from different Groups may share the same underlying storage, e.g.
+// networking.k8s.io ingresses and extensions ingresses. The StorageVersion manager
+// only needs to update one StorageVersion for the equivalent Groups.
+func dedupResourceInfos(infos []*ResourceInfo) []*ResourceInfo {
+	var ret []*ResourceInfo
+	seen := make(map[schema.GroupResource]struct{})
+	for _, info := range infos {
```

Comment:
We need to sort the infos to make sure the dedup always the pick the same GR to represent its equvalent resources, e.g., always pick network.k8s.io/ingress over extensions/ingress.